### PR TITLE
Cleanup code duplication from Ovirt provider

### PR DESCRIPTION
**Pull request by agrare**
We subclassed the Redhat Virtualization provider from the Ovirt provider, but a number of classes/methods were duplicated rather than use the parent class.

Sometimes this was an oversight, most times it was due to references directly to `Ovirt::InfraManager::...` which had to be changed to `Redhat::InfraManager::...` here.

These can be simplified by changing the parent class to work better with provider subclasses, for instance https://github.com/ManageIQ/manageiq-providers-ovirt/pull/625/files#diff-b3913d10eac6a2f99b4623a0a3b5212eb492a2a869c20770e4dee68cdf117ad5L72-R72

Depends:
- [x] https://github.com/ManageIQ/manageiq-providers-ovirt/pull/625

### CODE REVIEW

File: app/models/manageiq/providers/redhat/infra_manager.rb [hunk 1]
```diff
@@ -25,57 +25,17 @@ class ManageIQ::Providers::Redhat::InfraManager < ManageIQ::Providers::Ovirt::In
   require_nested  :DistributedVirtualSwitch
   include_concern :ApiIntegration
 
+  has_one :network_manager,
+          :foreign_key => :parent_ems_id,
+          :class_name  => "ManageIQ::Providers::Redhat::NetworkManager",
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/infra_manager.rb:30]: NOTE by declaring the class_name on the association here we don't have to override the `ensure_managers` method which was identical except for `build_network_manager(:type => 'ManageIQ::Providers::Redhat::NetworkManager')`

**Comment by agrare** [comment]
There is still more to go, but I don't want these PRs to be enormous.

Taking out of WIP

File: app/models/manageiq/providers/redhat/infra_manager.rb [hunk 2]
```diff
@@ -91,4 +51,20 @@ def self.catalog_types
...
     n_('Infrastructure Provider (Red Hat)', 'Infrastructure Providers (Red Hat)', number)
   end
+
+  def self.vm_vendor
+    "redhat".freeze
+  end
+
+  def self.host_vendor
+    "redhat".freeze
+  end
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/infra_manager.rb:61]: Can these 2 be moved up between description and catalog_types to match the ordering in the ovirt plugin?

File: app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb [hunk 3]
```diff
@@ -1,297 +1,9 @@
...
-
-  def get_timezones(_options = {})
-    SYSPREP_TIMEZONES
-  end
-
-  def self.default_dialog_file
-    'miq_provision_dialogs'
-  end
-
+class ManageIQ::Providers::Redhat::InfraManager::ProvisionWorkflow < ManageIQ::Providers::Redhat::InfraManager::ProvisionWorkflow
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb:1]: ? Is this right?

```suggestion
class ManageIQ::Providers::Redhat::InfraManager::ProvisionWorkflow < ManageIQ::Providers::Ovirt::InfraManager::ProvisionWorkflow
```

File: app/models/manageiq/providers/redhat/connection_manager.rb [hunk 4]
```diff
@@ -14,8 +14,6 @@
 # limitations under the License.
 #
 
-require 'ovirtsdk4'
-
 #
 # This class is a responsible for managing a set of `OvirtSDK4::Connection` objects. Each connection is
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/connection_manager.rb:18]: Is this comment needed anymore?  Seems like the base class has it covered.

**Comment by Fryguy** [comment]
Merged the core PR, but there is a conflict here.

File: app/models/manageiq/providers/redhat/infra_manager.rb (hunk 2 above)
Comment (agrare) [reply app/models/manageiq/providers/redhat/infra_manager.rb:61]: :+1: done, moved under `self.description`

File: app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb (hunk 3 above)
Comment (agrare) [reply app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb]: :+1: fixed

File: app/models/manageiq/providers/redhat/connection_manager.rb (hunk 4 above)
Comment (agrare) [reply app/models/manageiq/providers/redhat/connection_manager.rb:18]: :+1: removed

**Review by Fryguy — APPROVED** [review]
LGTM but tests are red

**Pull request merged**